Update environment.yml

BenjaminBossan opened the pull request:
For some reason, the old one didn't work for me anymore. I got the following errors:

```bash
$ conda --version
conda 4.7.10
$ conda env create
ResolvePackageNotFound: 
  - setuptools==27.2.0=py36_0
  - xz==5.2.2=1
  - zlib==1.2.8=3
  - readline==6.2=2
  - sqlite==3.13.0=0
  - openssl==1.0.2l=0
  - tqdm=4.14.0
  - tk==8.5.18=0
  - python==3.6.2=0
```

Could someone please confirm this?

If this is true, I wonder why this happens. Shouldn't conda allow reproducible builds? Were all those packages removed?

thomasjpfan commented:
Yea [conda removed some packages from their default/free channel](https://www.anaconda.com/why-we-removed-the-free-channel-in-conda-4-7/), if we want to have access to the old packages we need to run:

```bash
conda config --set restore_free_channel true
```

BenjaminBossan commented:
Well in that case we should probably update the `environment.yml`. I removed the third part of the package versions (`foo==bar==BAZ`) because I found that this prevents builds from being reproducible. I'm not sure if they are needed.

Can you verify that the updated `environment.yml` works?

thomasjpfan commented:
On OSX, I get this:

```
Collecting package metadata (repodata.json): done
Solving environment: failed

ResolvePackageNotFound: 
  - libgcc-ng=9.1.0
  - libstdcxx-ng=9.1.0
  - libgfortran-ng=7.3.0
```

Removing them worked.

BenjaminBossan commented:
> On OSX, I get this:

Do you mean with the new or the old environment.yml?

thomasjpfan commented:
New one in this PR

BenjaminBossan commented:
I updated the env by removing the offending lines. I also ticked up the default Python version to 3.7.

Additionally, I changed the unit tests to be compatible with the latest pytorch release (1.2).

BenjaminBossan commented:
I had to change the Python 3.7 version to 3.7.3 from 3.7.4 (default) because numpy couldn't be installed otherwise. Locally, I could replicate this by just creating an env with Python 3.7.4 and calling `pip install numpy`. Different recent numpy versions all had the same effect. Oddly, I couldn't find a solution for this, even though this should affect many people.

BenjaminBossan commented:
> With the adjustments for PyTorch 1.2, lets update the PYTORCH_VERSION in travis as well?

I would prefer to separate this and then immediately make new release.

BenjaminBossan commented:
@ottonemo What's your take?

ottonemo commented:
LGTM. I'm for merging this right now and making a separate PR for PyTorch 1.2 related changes and doing a release after that.
